torch/export/exported_program.py

```diff
@@ -354,9 +354,12 @@ def _is_joint_ir_decomp(ep, joint_loss_index):
         for node in mod.graph.nodes:
             if node.op == "placeholder":
                 if isinstance(node.meta["val"], CustomObjArgument):
-                    fake_args.append(
-                        maybe_to_fake_obj(fake_mode, node.meta["val"].fake_val.real_obj)  # type: ignore[union-attr]
-                    )
+                    real_script_obj = None
+                    if node.meta["val"].fake_val is None:
+                        real_script_obj = ep.constants[node.meta["val"].name]
```

Comment:
@ydwu4 so this happens when we trace "real" mode as a result the obj doesn't get fakified. Should we reflect this change here by just passing in real value instead of fake?

Reply:
Do you mean we always assign the real obj to CustomObjArgument.fake_val  (maybe rename to real_val) when we create a CustomObjArgument? Maybe it's OK. but not sure if other places are relying on CustomObjArgument.fake_val being Fake.

Reply:
Ah no. So the failing test is because we have a real mode of tracing as a result, we never fakified the obj. When we run_decomp, we will retrace the same real obj. So I am thinking whether I should just pass in real_script_obj to run_decomp instead of fakifying it before.

Reply:
I think I don't understand what's the implication here. If we run_decomp with real obj, that could potentially mutate the script object that might cause future re-tracing to fail? (e.g. pop from an empty tensor queue)